### DISCUSSION TITLE
Would pocketbase be a good choice for such project?

### DISCUSSION THREAD

**Discussion by mikekok** [pocketbase/pocketbase#1970] [Q&A]
I've used pocketbase for some small projects of mine and love everything about it.

Right now I'm trying to decide whether pocketbase would be a good option for a new project I'll be working on, mainly due to the fact that it uses sqlite.

The project is basically a sveltekit CRUD web app, that will have the following:
- Maximum of 10 concurrent users (it won't be public)
- Around 10 tables with 100.000 rows max each
- Less than 10GB of images uploaded as files
- Some sql join queries for returning statistics based on the tables' data (those won't be used more than 100 times per day)

Those 10 users will be constantly inserting data into the database, I'm guessing around 2000 new rows a day, all day long.
One of them, the admin, will use that data to get stats using the queries I mentioned above (nothing too complex, mostly joins, group bys and where clauses based on datetime).

So my dilemma is:
Should I go with pocketbase or supabase?
I know that supabase is more feature rich and scalable (which I don't think is an issue for my case) but I kinda wanna stick with pocketbase, thanks to the ease of use and production deployment (self hosting supabase seems to be intentionally difficult for some reason).

Will there be performance issues due to the fact that users will be adding new data concurrently, causing sqlite to constantly get locked, leading to slow performance and bad user experience?

I'd like to read your opinion on that, thank you for your time!

**Comment by ganigeorgiev** [comment]
> Will there be performance issues due to the fact that users will be adding new data concurrently, causing sqlite to constantly get locked, leading to slow performance and bad user experience?

For your particular use case (~10 concurrent users) I don't think that you'll ever encounter `SQLITE_BUSY`/`database is locked` errors. We use SQLite in [WAL mode](https://www.sqlite.org/wal.html) which allows non-blocking readers and writer.

Also, please note that in addition to the SQLite [busy_timeout](https://www.sqlite.org/pragma.html#pragma_busy_timeout) we have auto fail/retry for the `Dao` write and read (_since [v0.13.0](https://github.com/pocketbase/pocketbase/releases/tag/v0.13.0)_) operations.

In my opinion, your app is fairly small and PocketBase could handle it just fine.

**What you have to keep in mind is that PocketBase is still under active development and full backward compatibility is not guaranteed**!
Meaning that if you ended up choosing PocketBase, you have to be prepared to read the release notes or [CHANGELOG](https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md) and eventually perform some manual unavoidable upgrade steps from time to time.

**Reply by mikekok** [reply]
I didn't know that pocketbase supported non-blocking reading and writing, that's pretty awesome and actually dispels my performance concerns.

I'm aware that pocketbase is still under development and have no issue upgrading it from time to time.

Thanks for your time, keep up the great work!
